Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver/Registration.hs

```diff
@@ -204,8 +229,96 @@ underReviewDriversList _merchantShortId _opCity _limit _offset = throwError (Int
+        Common.UploadProfile imageId -> QImage.updateToValid True (Id imageId.getId)
+        Common.SSNApprove ssnNum -> do
+          ssnEnc <- encrypt ssnNum
+          QSSN.updateVerificationStatusAndReasonBySSN VerificationT.VALID Nothing (ssnEnc & hash)
+        Common.VehicleInsurance vInsuranceReq -> approveAndUpdateInsurance vInsuranceReq
+        Common.VehicleFitnessCertificate fitnessReq -> approveAndUpdateFitnessCertificate fitnessReq
+        Common.ProfilePhoto imageId -> QImage.updateToValid True (Id imageId.getId)
+        _ -> throwError (InternalError "Unknown Config in approve update document")
+    Common.Reject rejectReq -> do
+      case rejectReq of
+        Common.SSNReject ssnRejectReq -> rejectSSNAndSendNotification ssnRejectReq merchantOpCityId
+        Common.ImageDocuments imageRejectReq -> do
+          let imageId = Id imageRejectReq.documentImageId.getId
+          image <- QImage.findById imageId >>= fromMaybeM (InternalError "Image not found by image id")
+          case image.imageType of
+            Domain.ProfilePhoto -> QImage.updateIsValidAndFailureReason False (Just $ ImageNotValid imageRejectReq.reason) imageId
+            Domain.UploadProfile -> QImage.updateIsValidAndFailureReason False (Just $ ImageNotValid imageRejectReq.reason) imageId
+            Domain.DriverLicense -> QDL.updateVerificationStatus VerificationT.INVALID imageId
+            Domain.VehicleRegistrationCertificate -> QRC.updateVerificationStatus VerificationT.INVALID imageId
+            Domain.VehicleFitnessCertificate -> QFC.updateVerificationStatus VerificationT.INVALID imageId
+            Domain.VehicleInsurance -> QVI.updateVerificationStatus VerificationT.INVALID imageId
+            _ -> throwError (InternalError "Unknown Config in reject update document")
+          driver <- QDriver.findById image.personId >>= fromMaybeM (PersonNotFound image.personId.getId)
+          Notify.notifyDriver merchantOpCityId notificationType (notificationTitle (show image.imageType)) (message (show image.imageType)) driver driver.deviceToken
+  pure Success
+  where
+    notificationType = FCM.TRIGGER_SERVICE
+    notificationTitle obj = "Attention: Your " <> obj <> " is invalid."
+    message obj = "Kindly reapply or reupload your " <> obj
+    approveAndUpdateDL imageId = do
+      QDL.updateVerificationStatus VerificationT.VALID (Id imageId.getId)
+      QImage.updateToValid True (Id imageId.getId)
+    approveAndUpdateRC req = do
+      let imageId = Id req.documentImageId.getId
+      rc <- QRC.findByImageId imageId >>= fromMaybeM (InternalError "RC not found by image id")
+      let udpatedRC =
+            rc
+              { DRC.vehicleVariant = (castVehicleVariant <$> req.vehicleVariant) <|> rc.vehicleVariant,
+                DRC.verificationStatus = VerificationT.VALID
+              }
+      QRC.updateByPrimaryKey udpatedRC
+      QImage.updateToValid True imageId
+    approveAndUpdateInsurance req = do
```

Comment:
For insurance, fitness certificate we may need to create entry too in case it is not there if approved.